stuff2sort/System.py

```diff
@@ -0,0 +1,1150 @@
+        if input_dict['useinnerouter']:
+            self.xyzq = geometry.make_xyzq_io(self.initial_geometry, self.charge_vector, self.outeratomslist)
+        else:
+            self.xyzq = geometry.make_xyzq(self.initial_geometry, self.charge_vector)
+
+        #   Read linkatoms and next order atoms
+        self.m1list = self.identify_m1()
+        self.m2list = self.identify_m2()
+
+        self.linkatoms = get_linkatoms_ang(self.xyzq, self.qmatomslist, self.m1list, self.connectivity_list, [])
+        self.linkcorrlist, self.q1list, self.q2list, self.q3list, self.m3list = self.get_linkcorrlist()
+
+
+
+
+    def read_atoms_list(self, atoms_input):
+
+        '''
+        ------------------------------ \\
+        EFFECT: \\
+        ---------------
+        Reads Atom Indices
+        ------------------------------ \\
+        INPUT: \\
+        --------------- \\
+        atoms_input: str -> Index File \\
+        ------------------------------ \\
+        RETURN: \\
+        --------------- \\
+        atom_list: list -> List Of Atom Indices \\
+        ------------------------------ \\
+        '''
+
+        atom_list = []
+        with open(atoms_input) as atom_file:
+            for line in atom_file:
+                if "[" in line or "]" in line:
+                    continue
+                atom_index_list = re.findall("\d+", line)
+                if atom_index_list:
+                    atom_list.extend(map(int, atom_index_list))
+        atom_list = sorted(np.array(atom_list).astype(int))
+        return atom_list
+
+    def read_molecules(self):
+
+        '''
+        ------------------------------ \\
+        EFFECT: \\
+        --------------- \\
+        Reads List Of Molecules From The Topology File \\
+        ------------------------------ \\
+        INPUT: \\
+        --------------- \\
+        NONE \\
+        ------------------------------ \\
+        RETURN: \\
+        --------------- \\
+        molecule_list: list -> List Of Molecules And Their Amount In The System \\
+        ------------------------------ \\
+        '''
+
+        molecule_list = []
+        with open(self.input_dict['topologyfile']) as ifile:
+            found = False
+            for line in ifile:
+                match = re.search(r"^\[ molecules \]", line, flags=re.MULTILINE)
+                if match:
+                    found = True
+                    break
+            if not found:
+                #   XX AJ turn into Logging
+                print('No "molecules" entry in ' + str(self.top) + " found. Exiting.")
+                exit(1)
+            for line in ifile:
+                match = re.search(r"^;", line, flags=re.MULTILINE)
+                if match:
+                    continue
+                else:
+                    match = re.search(r"^(\S+)\s+(\d+)", line, flags=re.MULTILINE)
```

Comment:
Especially when working with Regex, try to formulate in a comment what you want to achieve, as regex is hard to read and one needs to remember or look up the symbolics. It helps to get an idea of the code more quickly.

Reply:
I understand the benefit, I just want to mention that it will take a long time to do that for the whole code. We can discuss that in the meeting today.

Reply:
# Find All Molecule Types And Their Amount
                else:
                    # Extract Lines With A String (Molecule Type, Group 1) And A Number (Molecule Amount, Group 2) Seperated By Whitespace
                    match = re.search(r"^(\S+)\s+(\d+)", line, flags=re.MULTILINE)
                    if match:
                        list_molecule.append([match.group(1), match.group(2)])

This is how I would comment this example. Is that alright?